src/utils/userApi.js

```diff
@@ -1 +1,113 @@
-// TO BE COMPLETED WITH BACKEND PART OF THE PROJECT //
+/* eslint-disable no-console */
+class UserApi {
+  constructor({ baseUrl }) {
+    this._baseUrl = baseUrl;
+  }
+
+  _getResponseData(res) {
+    if (!res.ok) {
+      return Promise.reject(res);
+    }
+    return res.json();
+  }
+
+  registerUser(userData) {
+    return fetch(`${this._baseUrl}/users/register`, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        email: userData.email,
+        password: userData.password,
+        name: userData.name
+      })
+    }).then((res) => this._getResponseData(res));
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._getResponseData)
  }

```

And now you can safely replace `fetch` with `this._request`. And you don’t need to change anything else.